ANTLR/src/Compiler/ContextualAnalysis/TypeCheckerVisitor.java

```diff
@@ -182,10 +194,10 @@ public Integer visitValId(ValIdContext ctx) {
      * @return The return type of the function definition.
      */
     @Override
-    public Integer visitFuncdef(FuncdefContext ctx) {
+    public Integer visitFuncDef(FuncDefContext ctx) {
         // Check return statement's return type
-        Integer returnStmtType = visit(ctx.stmt());
-        checkReturnTypeCorrespondence(returnStmtType, ctx, ctx.stmt());
+        Integer returnStmtType = visit(ctx.returnStmt());
+        checkReturnTypeCorrespondence(returnStmtType, ctx, ctx.returnStmt().stmt());
```

Comment:
Måske vi kunne visit returnStmt inde i checkReturnType...(), så slipper vi for den første linje her, samt for en parameter i funktionskaldet?

Reply:
Den bliver brugt to forskellige steder med forskellig sammenligningsparameter. Jeg har derfor fjerne returnsStmtType og sat visit ind som parameter i stedet

Reply:
Vi bruger returnStmt to steder. bare i kaldet og som returværdi, så alternativet er at rykke visit ind som parameter og så returnerer visit igen. Synes det er bedre at det ligger i en variable så vi slipper for to funktionskald. Derudover er første parameter ikke det samme for hvert kald, da den længere nede bliver brugt på formelle vs aktuelle parametre. Af den grund har jeg ændret funktionsnavn til "checkTypeCorrespondence"